src/Management/src/Endpoint/HeapDump/ConfigureHeapDumpEndpointOptions.cs

```diff
@@ -3,16 +3,33 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Steeltoe.Common;
 using Steeltoe.Management.Endpoint.Options;
 
 namespace Steeltoe.Management.Endpoint.HeapDump;
 
 internal sealed class ConfigureHeapDumpEndpointOptions : ConfigureEndpointOptions<HeapDumpEndpointOptions>
 {
+    private readonly ILogger<ConfigureHeapDumpEndpointOptions> _logger;
     private const string ManagementInfoPrefix = "management:endpoints:heapdump";
 
-    public ConfigureHeapDumpEndpointOptions(IConfiguration configuration)
+
+    public ConfigureHeapDumpEndpointOptions(IConfiguration configuration, ILogger<ConfigureHeapDumpEndpointOptions> logger)
         : base(configuration, ManagementInfoPrefix, "heapdump")
     {
+        _logger = logger;
+    }
+
+    public override void Configure(HeapDumpEndpointOptions options)
+    {
+        base.Configure(options);
+
+        // Full dumps are broken on Osx, so default to gcdump
+        if (Platform.IsOSX)
+        {
+            _logger.LogInformation("Full dumps are not supported on OSX, defaulting to gcdump.");
```

Comment:
```suggestion
        // Full dumps are broken on OSX, so only allow gcdump
        if (Platform.IsOSX && options.HeapDumpType != "gcdump")
        {
            _logger.LogInformation("Only GC dumps are supported on OSX.");
```

We're not just "defaulting" if we're forcing it to always be the same. According to the docs, we allow several types beyond full and gc, have you tested the others?

> Acceptable values are Normal, WithHeap, Triage and Full.

Apparently we need to add gcdump to the docs too